lib/AST/ArchetypeBuilder.cpp

```diff
@@ -2028,7 +2028,10 @@ Type ArchetypeBuilder::mapTypeIntoContext(Module *M,
   if (!canType->hasTypeParameter())
     return type;
 
-  assert(genericParams && "dependent type in non-generic context");
+  if (!genericParams) {
+    // Failed: dependent type in non-generic context.
+    return Type();
+  }
```

Comment:
09650-swift-typebase-getcanonicaltype.swift

From the fuzzer, so it's not valid code anyway.